### PR TITLE
I made controller...

### PR DESCRIPTION
**Pull request by mwames**

### CODE REVIEW

File: src/controller/controller.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    return true;
+}
+
+void milk::controller_close()
+{
+    SDL_GameControllerClose(controller);
+    controller = NULL;
+}
+
+void milk::controller_handle_axis_event()
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/controller/controller.cpp]: Yeah, I was thinking about that last night

File: CMakeLists.txt [hunk 2]
```diff
@@ -176,6 +195,7 @@ target_link_libraries(milk
 # assumes SDL2_image 2.0.4
 # assumes Lua 5.3.5
 #---------------------------------------------------------------------------
+set_target_properties(milk PROPERTIES PREFIX "")
```
[the remark this answers is not in the document]
Comment (mwames) [reply CMakeLists.txt]: By default in a unix system cmake prefixes with "lib" so I was getting libmilk.so which doesn't work so well when we're importing "milk" so this just gets rid of the lib part.

File: src/controller/controller.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+bool milk::controller_init() 
+{
+    if(SDL_Init(SDL_INIT_GAMECONTROLLER) < 0)
+    {
+        std::cout << "SDL_Init: Failed to initialize gamepad: " << SDL_GetError() << std::endl;
+        return false;
+    }
+
+    controller = SDL_GameControllerOpen(0);
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/controller/controller.cpp]: Nope, there are events we can listen for, just not implemented yet.

File: src/controller/controller.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    void controller_close();
+    void controller_handle_axis_event();
+    //TODO: return normalized axes values;
+    int controller_get_leftx();
+    int controller_get_lefty();
+    int controller_get_rightx();
+    int controller_get_righty();
+    //ENDTODO
+    void controller_update_state();
+    bool controller_is_button_down(SDL_GameControllerButton button);
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/controller/controller.h]: I can address that soon

File: src/milk_export.h [hunk 5]
```diff
@@ -1,6 +1,10 @@
 #ifndef _MILK_EXPORT_H_
 #define _MILK_EXPORT_H_
 
-#define MILK_EXPORT __declspec(dllexport)
+    #if defined(_MSC_VER)
+        #define MILK_EXPORT __declspec(dllexport)
+    #elif defined(__GNUC__)
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/milk_export.h]: is tight

File: src/luaopen_milk.cpp [hunk 6]
```diff
@@ -11,14 +11,20 @@ extern "C" {
...
 static int milk_init(lua_State* L)
 {
 	if (initialized)
 		return 0;
-	if (milk::window_init() && milk::graphics_init(milk::window_get_handle()) && milk::audio_init()) {
+	if (milk::window_init()
+		&& milk::graphics_init(milk::window_get_handle())
+		&& milk::audio_init()
+		&& milk::joystick_init()
+		&& milk::controller_init()) {
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/luaopen_milk.cpp]: It's my first day!

File: src/window/window.cpp [hunk 7]
```diff
@@ -58,10 +60,15 @@ void milk::window_poll()
 			milk::window_close();
 		if (event.type == SDL_MOUSEWHEEL)
 			milk::mouse_handle_wheel_event(&event);
+		if (event.type == SDL_JOYAXISMOTION)
+			milk::joystick_handle_axis_event(&event);
+		if (event.type == SDL_CONTROLLERAXISMOTION)
+			milk::controller_handle_axis_event();
```
[the remark this answers is not in the document]
Comment (mwames) [reply src/window/window.cpp]: Yep, I was being lazy

File: src/luaopen_milk.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (Straskal) [reply src/luaopen_milk.cpp]: RIP nick

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Straskal) [reply CMakeLists.txt]: +1

**Pull request merged**